interpreter/valid/valid.ml

```diff
@@ -177,15 +177,26 @@ let check_memop (c : context) (memop : 'a memop) get_sz at =
   require (1 lsl memop.align <= size) at
     "alignment must not be larger than natural"
 
-let check_simd_lane_idx op at =
+let check_simd_extract_lane_index op at =
```

Comment:
This duplication is slightly annoying, due to the chosen representation for `extractop` and `replaceop`. Is there way to align them? Maybe using a record or something?

Reply:
Agree, it would be nicer to combine these. I don't know ocaml well enough to say how that could be done, though

Reply:
It should work to make it a higher-order function taking an extractor function:
```
let check_simd_lane_index f op at =
  let max, op' = match op with
    | V128Op.I8x16 op' -> 16, op'
    | ...
  in require (f op' < max) at "invalid lane index"
```
and call it with `check_simd_lane_index Fun.id` vs `check_simd_lane_index snd`.

Reply:
Great! Made that change, thanks!